lib/link.js

```diff
@@ -26,6 +26,12 @@ export default class Link extends Component {
     const route = as ? href : null
     const url = as || href
 
+    //  avoid scroll for urls with anchor refs
+    let { scroll } = this.props
+    if (url.indexOf('#') > -1 && !scroll) {
```

Comment:
@nkzawa what are your thoughts on this? We would only update if `scroll` is a falsey value. That way it will cover the scenario where no `scroll` prop is being passed in but still allow for users to pass in `scroll={true}` in order to  maintain the auto scroll- even if their url includes `#`